### ISSUE TITLE
Raspberry pi 3 can't go above 4hz

### ISSUE THREAD

**Issue by Yours3lf** [pimoroni/mlx90640-library#60]
Hi,

I'm trying to run test.cpp at frame rates above 4hz, but it just gives me the "cnt > 4" error for any value above 4hz.
4hz does work though.
I'm using a raspberry pi 3 B+ with 1Mhz baudrate and the default i2c mode (linux)
the camera module is the pimoroni MLX90640 breakout from pihut (PIM365)
i2cdetect gives me 0x33
it is the only thing connected to the RPi.
I basically wired up
pi -> mlx90640
3v3 to 3.3v-6v
sda to sda
scl to scl
gnd to gnd

How can you run this at the advertised 64hz?

thanks!

**Comment by Gadgetoid** [comment]
High-frequency i2c can be really sensitive to wiring, do you have a picture of your setup?

**Comment by Yours3lf** [comment]
![Uploading 4FD7D9EE-9423-4750-B335-E3B433B20EA1.jpeg…]()

**Comment by Yours3lf** [comment]
![72694816-9B1C-4986-AEF6-1BBF173D3E2E](https://user-images.githubusercontent.com/1737843/185993009-3844a498-7fc3-4317-a724-b1e5bafe1bbf.jpeg)
![96AC5F59-7CB0-4E8F-882F-DC80ACAFE527](https://user-images.githubusercontent.com/1737843/185993020-433a3fbe-d129-4419-a67d-76fb2ea57546.jpeg)

**Comment by Yours3lf** [comment]
![6DE4A320-0958-4590-A536-99ACF278BDE2](https://user-images.githubusercontent.com/1737843/185993372-3199fad8-2993-4757-90ba-5e3a8865a2e9.jpeg)

**Comment by Gadgetoid** [comment]
Try taking the ribbon cable out of the mix and wiring some jump wires directly to the sensor. Ideally from pins on the Pi to pins on the sensor itself and not via a breadboard.

**Comment by Yours3lf** [comment]
Okay I'll try that

**Comment by Yours3lf** [comment]
![IMG_5867](https://user-images.githubusercontent.com/1737843/188211025-bc683704-e02a-439e-8ac9-fcd4355cc769.JPEG)
![IMG_5868](https://user-images.githubusercontent.com/1737843/188211029-adcfa126-fd12-4456-875b-7099c943641e.JPEG)
![IMG_5869](https://user-images.githubusercontent.com/1737843/188211038-428d7c51-8a56-4b38-82f4-0c3722e856e8.JPEG)

**Comment by Yours3lf** [comment]
hey, tried it today, but the above setup has the same behaviour unfortunately.

**Comment by Yours3lf** [comment]
hey, I'd like to bump this issue please

**Comment by Yours3lf** [comment]
any other suggestion please?

**Comment by myslide** [comment]
I build it according the instructions on a Rpi 400. I changed the config.txt. without a reboot and got these error too in ./fbuf. Rebooted, make clean, make I2C_MODE ->./fbuf confirmed the success by 1440x900 but none picture-> Ok, resolution is to hight. The ./test shows the raw pixels with a slow framerate. But as I tried  sdlsclae it satisfied me with the acceptable interpolation and framerate. So these are my experiences. I wired it up with 10cm reused, preconfigured plug and soldered it on the breakout board. Make sure your system is up to date, the required libs are installed (I got warnings with the libavutils reg. the decoder), make clean, make all... . Keep tought!
